### PR TITLE
Remove MySQL and RabbitMQ

### PR DESCRIPTION
**Pull request by Maffooch**
Following the deprecation of MySQL and RabbitMQ ([see discussion post here](https://github.com/DefectDojo/django-DefectDojo/discussions/9690)), it is now time for removal. The following places have been touched:
- Unit tests
  - Remove matrix values for testing mysql/rabbitmq
  - Remove the `--profile` flag from other tests using `dc-` helper scripts
- Simplify docker compose to remove the need for profiles
- Remove all docker profile environments
  - This includes setting default values in the compose file for all Postgres/redis related env vars 
- Clean up the docs to remove suggestions that mysql/rabbitmq are supported/configurable
- Remove mysql/rabbitmq traces from the helm chart
- Add breaking changes to 2.37.0 release notes 

[sc-7056]

### CODE REVIEW

**Comment by Maffooch** [comment]
Converting to draft until I get to the bottom of rest test framework failures

**Review by mtesauro — APPROVED** [review]
Approved

**Review by blakeaowens — APPROVED** [review]

**Review by hblankenship — COMMENTED** [review]
I note that there is a comment from gitbot regarding the following:

Check warning on line 461 in helm/defectdojo/values.yaml

GitHub Actions
/ Lint chart

461:12 [truthy] truthy value should be one of [false, true]

**Review by cneill — COMMENTED** [review]
A few comments on this one, and a couple files that weren't totally purged of MySQL/RabbitMQ references yet:

* We can yank [these lines ignoring MySQL/RabbitMQ](https://github.com/DefectDojo/django-DefectDojo/blob/master/.github/renovate.json#L10-L13) out of `renovate.json` now
* This [gory bit](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/metrics/utils.py#L534-L540) of `dojo/metrics/utils.py`
* [This bit in settings.dist.py](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/settings/settings.dist.py#L1633-L1635)
* The [mysqlclient blob in NOTICE](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/NOTICE#L6431-L6772)

There are some other comments about MySQL-specific behavior scattered in some other files but we can worry about that later - I don't think it will hurt anything to leave them alone for now.

File: docker-compose.yml [hunk 1]
```diff
@@ -127,47 +102,20 @@ services:
...
-       - defectdojo_rabbitmq:/var/lib/rabbitmq
   redis:
     image: redis:7.2.5-alpine@sha256:0bc09d9f486508aa42ecc2f18012bb1e3a1b2744ef3a6ad30942fa12579f0b03
-    profiles: 
-      - mysql-redis
-      - postgres-redis
     volumes:
       - defectdojo_redis:/data
 volumes:
   defectdojo_data: {}
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply docker-compose.yml:118]: Looks like it can go

**Review by hblankenship — APPROVED** [review]

**Comment by Maffooch** [comment]
> - This [gory bit](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/metrics/utils.py#L534-L540) of dojo/metrics/utils.py
> - [This bit in settings.dist.py](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/settings/settings.dist.py#L1633-L1635)

I would really rather not touch these to be honest 😅

**Comment by cneill** [comment]
> > * This [gory bit](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/metrics/utils.py#L534-L540) of dojo/metrics/utils.py
> > * [This bit in settings.dist.py](https://github.com/DefectDojo/django-DefectDojo/blob/97c47c874e330f7b943e74664a968d0b3f44e4fb/dojo/settings/settings.dist.py#L1633-L1635)
> 
> I would really rather not touch these to be honest 😅 @cneill

No problem, just thought I'd mention them since ~everything else is cleaned up and we might forget about them. But we're probably already doing enough with this PR.

**Review by cneill — APPROVED** [review]

**Pull request merged**